sdk/rust/oak_utils/src/lib.rs

```diff
@@ -14,159 +14,165 @@
-}
+            #[allow(dead_code)]
+            pub struct #dispatcher_name<T: #service_name>(T);
 
-impl FileDiff {
-    // Returns a file action based on whether file contents were changed or not.
-    fn action(&self) -> FileAction {
-        match (self.old_content.as_ref(), self.new_content.as_ref()) {
-            (Some(old_content), Some(new_content)) => {
-                if old_content == new_content {
-                    FileAction::Ignore
-                } else {
-                    FileAction::Update {
-                        new_content: new_content.to_string(),
-                    }
+            impl <T: #service_name> #dispatcher_name<T> {
+                pub fn new(node: T) -> #dispatcher_name<T> {
+                    #dispatcher_name(node)
                 }
             }
-            (Some(_), None) => {
-                // `mod.rs` file is not generated by `protoc_rust` and `protoc_rust_grpc`.
-                if "mod.rs" == self.filename {
-                    FileAction::Ignore
-                } else {
-                    FileAction::Remove
+
+            #[allow(clippy::unit_arg)]
+            impl <T: #service_name> #oak_package::grpc::ServerNode for #dispatcher_name<T> {
+                fn invoke(&mut self, method: &str, req: &[u8], writer: #oak_package::grpc::ChannelResponseWriter) {
+                    match method {
+                        #(#dispatcher_methods,)*
+                        _ => {
+                            ::log::error!("unknown method name: {}", method);
+                        }
+                    }
                 }
             }
-            (None, Some(new_content)) => FileAction::Update {
-                new_content: new_content.to_string(),
-            },
-            _ => FileAction::Ignore,
-        }
-    }
-}
 
-// Copies changed files from `temp_path` to `path` and removes files from `path`
-// that are not present in `temp_path`.
-fn update_files(temp_path: &Path, path: &Path) -> io::Result<()> {
-    let files = get_files(path);
-    let temp_files = get_files(temp_path);
-    get_file_diffs(&files, &temp_files)
-        .iter()
-        .try_for_each(|diff| {
-            let filepath = path.join(&diff.filename);
-            match diff.action() {
-                FileAction::Ignore => Ok(()),
-                FileAction::Remove => fs::remove_file(filepath),
-                FileAction::Update { new_content } => {
-                    // `File::create` truncates an existing file.
-                    let mut file = fs::File::create(filepath).unwrap();
-                    file.write_all(new_content.as_bytes())
-                }
+            #[allow(dead_code)]
+            pub struct #client_name(pub #oak_package::grpc::client::Client);
+
+            #[allow(dead_code)]
+            impl #client_name {
+                #(#client_methods)*
             }
-        })
+        };
+        out.push_str("\n");
+        out.push_str(&format!("{}", gen));
```

Comment:
The output from this is all squished onto a single line in the generated code; is there an easy way to get it properly formatted (e.g. pipe it through ` rustfmt_nightly::Session::format`)?  (Not worth holding this up if it's not straightforward, though).

Reply:
I thought about that, but I didn't want to just pull in rustfmt along with this, it's quite a large dependency. But I'll leave a TODO for now.